src/app/(food)/allergens.tsx

```diff
@@ -33,10 +62,43 @@ export default function FoodPreferences(): JSX.Element {
         useContext(FoodFilterContext)
     filteredAllergens.sort((a, b) => a.title.localeCompare(b.title))
 
+    const navigation = useNavigation()
+
+    useEffect(() => {
+        navigation.setOptions({
+            headerSearchBarOptions: {
+                placeholder: t('navigation.allergensSearch', {
+                    ns: 'navigation',
+                }),
+
+                ...Platform.select({
+                    android: {
+                        headerIconColor: colors.text,
+                        textColor: colors.text,
+                        hintTextColor: colors.text,
+                    },
+                }),
+                shouldShowHintSearchIcon: false,
+                hideWhenScrolling: false,
+
+                onChangeText: (event: { nativeEvent: { text: string } }) => {
+                    router.setParams({
```

Comment:
as the search input moved to the component itself, maybe a simple react state is more readable the router params? (just a suggestion, idk if possible)